### DISCUSSION TITLE
for the love of god, how to effectively white label while keeping code dry

### DISCUSSION THREAD

**Discussion by binaryartifex** [nrwl/nx#17811] [General]
As the title implies, im at my wits end trying to create a white label application that doesn't involve needless repetition. i have a react application, lets just call it App, which is essentially a shell who's content is derived by a myriad of libraries. however, i want to be able to change the theme, manifest, images (background images), logos and icons by passing an env variable which grabs all these things from lib/brand-a or lib/brand....z. there could be an arbitrary number of themes/brands. 

ive gotten as far as isolating the tailwind configs, pre generated manifests, splash screens, images, logos, icons, etc into the brand libraries, however i cannot, for the life of me, figure out how to CONDITIONALLY import the one brand i want. i can use Vite's static asset copy plugin to get the build stuff (manifest, splash screens, etc) however i need to inject the assets (images, icons) into an asset provider so it has to be provided on the react side of the app, not the build side. and i don't want to create a different shell for each app as i just want to create a single e2e testing app to go with a single white labelled app. right now im forced to import all of the brands and conditionally select one based on the env variable but because they're all imported, all their assets are included in the final build. would be a lot easier if we could resolve env variables in the project.json file but i digress....any help would be much appreciated....

**Comment by cblaze22** [comment]
Ever figure this out?
